### PR TITLE
[FEATURE] Export isotope errors when reading pepXML

### PR DESCRIPTION
**Pull request by jpfeuffer**
# Description

For use in Percolator, parse isotope error with round(neutralmassdiff/C12C13diff).
Also contains various fixes for terminal modifications once again.

Contains another try to make clangd work out of the box on Gitpod.

### CODE REVIEW

File: src/openms/include/OpenMS/FORMAT/PepXMLFile.h [hunk 1]
```diff
@@ -296,6 +296,9 @@ namespace OpenMS
     /// RT and m/z of current PeptideIdentification (=spectrum)  
     double rt_{}, mz_{};
 
+    /// possible isotopes to consider and annotate
+    //std::vector<int> potential_isotopeerrors_;
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/FORMAT/PepXMLFile.h:300]: Unused?

File: src/openms/source/FORMAT/PepXMLFile.cpp [hunk 2]
```diff
@@ -1296,6 +1317,10 @@ namespace OpenMS
           peptide_hit_.setMetaValue("num_matched_peptides", value);
         }
       }
+
+      double pepmassdiff = attributeAsDouble_(attributes, "massdiff");
```
Comment (timosachsenberg) [inline src/openms/source/FORMAT/PepXMLFile.cpp:1321]: Is this mandatory or can it be optional in pepXML?

**Review by timosachsenberg — APPROVED** [review]
Maybe double check if we use the sign of the isotope error the same through openms

**Comment by jpfeuffer** [comment]
The only time we use it is here: https://github.com/OpenMS/OpenMS/blob/develop/src/topp/PercolatorAdapter.cpp#L501
where it must be signed to yield correct results (if I am not mistaken).
The only other method that annotates it is MSGFPlus. I don't know if it can be negative there.

File: src/openms/source/FORMAT/PepXMLFile.cpp (hunk 2 above)
Comment (jpfeuffer) [reply src/openms/source/FORMAT/PepXMLFile.cpp:1321]: required

```
<xs:attribute name="massdiff" type="xs:string" use="required">
<xs:annotation>
<xs:documentation>Mass(precursor ion) - Mass(peptide)</xs:documentation>
</xs:annotation>
</xs:attribute>
```

File: src/openms/include/OpenMS/FORMAT/PepXMLFile.h (hunk 1 above)
Comment (jpfeuffer) [reply src/openms/include/OpenMS/FORMAT/PepXMLFile.h:300]: yes

**Comment by jpfeuffer** [comment]
I fixed the MyriMatch test errors. Problem is, that the heuristic does not work since Myrimatch calculates/reports wrong theoretical peptide masses:
```
    <spectrum_query spectrum="spectra.3.3.3" spectrumNativeID="spectrum=2" start_scan="3" end_scan="3" precursor_neutral_mass="2323.13979169311" assumed_charge="3" index="3" retention_time_sec="4923.77734375002">
      <search_result num_target_comparisons="1" num_decoy_comparisons="0">
        <search_hit hit_rank="1" peptide="RPGADSDIGGFGGLFDLAQAGFR" peptide_prev_aa="-" peptide_next_aa="A" protein="BSA3" num_tot_proteins="1" calc_neutral_pep_mass="2324.50666" massdiff="1.36686830689" num_tol_term="2" num_missed_cleavages="1" num_matched_ions="24" tot_num_ions="46">
```

calc_neutral_pep_mass should be 2323.13.. I have no idea what they are doing..

**Comment by jpfeuffer** [comment]
If we disable isotopes in myrimatch it still finds them, so it is a problem of reporting.
What do we do? Calculate on our own? I guess so.

**Comment by timosachsenberg** [comment]
Myrimatch seems to be not developed further so don’t worry

**Pull request merged**